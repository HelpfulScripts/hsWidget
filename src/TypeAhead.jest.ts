import * as hswidget from './';
import { m }    from 'hslayout';

window = Object.assign(require('mithril/test-utils/domMock.js')(), require('mithril/test-utils/pushStateMock')());
const mq = require('mithril-query');

let hero = '';
const out = mq(m(hswidget.TypeAhead, { desc: {
    placeholder: 'favorite hero',
    onsubmit: item => hero = item,
    onkeydown: console.log,
    list: ['Batman', 'Superman', 'Spiderman', 'Hulk']
}}));

const myEvent = {
    target: {
        value:'result',
        setSelectionRange: (n, length) => {}
    },
    code:''
};

describe('RadioButton', () => {
    it ('should have DOM structure', () => {
        out.should.have('.hs-form');
        out.should.have('.hs-form input.hs-typeahead-input');
    });
    it('should type S', () => {
        myEvent.code = 'KeyS';
        out.keydown('.hs-form input', 83, myEvent);
        myEvent.code = 'Enter';
        out.keydown('.hs-form input', 13, myEvent);
        expect(hero).toBe('');
    });
});