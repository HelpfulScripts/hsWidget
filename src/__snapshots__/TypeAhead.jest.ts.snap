// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Ahead should match 1`] = `
Object {
  "blur": [Function],
  "click": [Function],
  "contains": [Function],
  "contextMenu": [Function],
  "find": [Function],
  "first": [Function],
  "focus": [Function],
  "has": [Function],
  "keydown": [Function],
  "keypress": [Function],
  "keyup": [Function],
  "log": [Function],
  "mousedown": [Function],
  "mouseenter": [Function],
  "mouseleave": [Function],
  "mouseout": [Function],
  "mouseover": [Function],
  "mouseup": [Function],
  "onremove": [Function],
  "onremovers": Array [],
  "redraw": [Function],
  "rootNode": Object {
    "attrs": Object {
      "className": "hs-form",
    },
    "children": Array [
      Object {
        "attrs": Object {
          "autofocus": true,
          "className": "hs-typeahead-input hs-typeahead-value",
          "contenteditable": true,
          "oninput": [Function],
          "onkeydown": [Function],
          "placeholder": "favorite hero",
        },
        "children": Array [
          Object {
            "attrs": undefined,
            "children": "result",
            "dom": undefined,
            "domSize": undefined,
            "events": undefined,
            "instance": undefined,
            "key": undefined,
            "parent": [Circular],
            "renderedChildren": "result",
            "state": undefined,
            "tag": "<",
            "text": undefined,
          },
        ],
        "dom": undefined,
        "domSize": undefined,
        "events": undefined,
        "instance": undefined,
        "key": undefined,
        "parent": [Circular],
        "renderedChildren": Array [
          Object {
            "attrs": undefined,
            "children": "result",
            "dom": undefined,
            "domSize": undefined,
            "events": undefined,
            "instance": undefined,
            "key": undefined,
            "parent": [Circular],
            "renderedChildren": "result",
            "state": undefined,
            "tag": "<",
            "text": undefined,
          },
        ],
        "state": undefined,
        "tag": "input",
        "text": undefined,
      },
      null,
    ],
    "dom": undefined,
    "domSize": undefined,
    "events": undefined,
    "instance": undefined,
    "key": undefined,
    "parent": undefined,
    "renderedChildren": Array [
      Object {
        "attrs": Object {
          "autofocus": true,
          "className": "hs-typeahead-input hs-typeahead-value",
          "contenteditable": true,
          "oninput": [Function],
          "onkeydown": [Function],
          "placeholder": "favorite hero",
        },
        "children": Array [
          Object {
            "attrs": undefined,
            "children": "result",
            "dom": undefined,
            "domSize": undefined,
            "events": undefined,
            "instance": undefined,
            "key": undefined,
            "parent": [Circular],
            "renderedChildren": "result",
            "state": undefined,
            "tag": "<",
            "text": undefined,
          },
        ],
        "dom": undefined,
        "domSize": undefined,
        "events": undefined,
        "instance": undefined,
        "key": undefined,
        "parent": [Circular],
        "renderedChildren": Array [
          Object {
            "attrs": undefined,
            "children": "result",
            "dom": undefined,
            "domSize": undefined,
            "events": undefined,
            "instance": undefined,
            "key": undefined,
            "parent": [Circular],
            "renderedChildren": "result",
            "state": undefined,
            "tag": "<",
            "text": undefined,
          },
        ],
        "state": undefined,
        "tag": "input",
        "text": undefined,
      },
      "",
    ],
    "state": undefined,
    "tag": "div",
    "text": undefined,
  },
  "setValue": [Function],
  "should": Object {
    "contain": [Function],
    "have": [Function],
    "not": Object {
      "contain": [Function],
      "have": [Function],
    },
  },
  "trigger": [Function],
}
`;
